### ISSUE TITLE
thread 'main' panicked at '[calloop] Encountered an orphan event: wl_keyboard@19 

### ISSUE THREAD

**Issue by twitchyliquid64** [l4l/yofi#94]
Repro:

1. Clone at head (rn thats https://github.com/l4l/yofi/commit/17edf903cc76f9e83e7f77ba39de87fa26c44a4d)
2. `LD_LIBRARY_PATH="$(nix eval --raw nixpkgs.wayland)/lib" cargo run`

Crash:
```
thread 'main' panicked at '[calloop] Encountered an orphan event: wl_keyboard@19 : keymap', /home/xxx/.cargo/registry/src/github.com-1ecc6299db9ec823/smithay-client-toolkit-0.15.3/src/event_loop.rs:49:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[wayland-client error] A handler for wl_keyboard panicked.
Aborted (core dumped)
```

(The ld-preload nonsense is because im testing on nixOS - this lets the program find the wayland libraries on my system).

**Comment by l4l** [comment]
This one seems related: https://github.com/Smithay/wayland-rs/issues/372

**Comment by l4l** [comment]
Should be fixed by #138
